osu.Game.Rulesets.Mania/Difficulty/Skills/README.md

```diff
@@ -0,0 +1,280 @@
+# Mania Strain Calculation Primer
+
+`StrainValueOf` governs how note strain is evaluated, which in-turn affects how star rating is calculated.
+
+---
+
+## Theory
+
+Strain refers to the difficulty of the note.
+
+There are 3 strain values present.
+
+- **Global Strain** $GS$: the overall running strain, depends on all previous notes.
+- **Column Strain** $CS$: the per-column running strain, depends on all previous notes in the same column.
+  - $CS_k$: the $k^{th}$ column running strain. 
+- **Strain** $S$: a function of GS and CS. $S=S(GS, CS)$
+
+### Intuition
+
+Take for example a simple strain calculator:
+
+- A 2K Map denoted on the columns K0, K1, with 3 notes, marked by X
+- Each note increments both strains by 5
+- Each time step decays both strain by 1 till 0
+- Strain is the sum of Global and the note's column-th Column Strain
+
+| Time | GS | CS0 | CS1 | S    | K0 | K1 | Desc.                                   |
+|------|----|-----|-----|------|----|----|-----------------------------------------|
+| -1   | 1  | 0   | 0   |      |    |    | Initialize Strains                      |
+| 0    | 6  |     | 5   | 6+5  |    | X  | 1st Note on Column 1, add to GS and CS1 |
+| 1    | 5  |     | 4   |      |    |    | Apply Decay                             |
+| 2    | 10 | 5   | 3   | 10+5 | X  |    | 2nd Note on Column 0, add to GS and CS0 |
+| 3    | 9  | 4   | 2   |      |    |    | Apply Decay                             |
+| 4    | 14 | 3   | 7   | 14+7 |    | X  | 3rd Note on Column 1, add to GS and CS1 |
+| 5    | 13 | 2   | 6   |      |    |    | Apply Decay                             |
+
+- GS increments on every note regardless of column, while CS only increments for their respective column.
+- S depends on GS and CS, never itself.
+    - On the 2nd note, S=GS+CS0 ignoring CS1 because the note's column is 0
+    - Same goes for the 3rd note, S=GS+CS1
+- CS is a matrix, GS is a vector.
+
+Though the actual script is non-trivial, it has similar concepts from the above example:
+
+- Our decay is [exponential](#evaluating-strains), so high strain values are exponentially rarer
```

Comment:
What does "high strain values are exponentially rarer" even mean? That's not even rigorous. I don't know what the probability distribution of strains is. Just leave it at the "decay is exponential" part.